reasoner/controller/ControllerRegistry.java

```diff
@@ -215,6 +216,9 @@ private static Map<Variable.Retrievable, Variable.Retrievable> identity(Resolvab
     }
 
     private Optional<ControllerView.MappedConcludable> getConcludable(Concludable concludable) {
+        // TODO: Due to a lack of synchronisation,
+        //      this does not guarantee that all alpha-equivalent concludables will re-use the same controller.
+        //      We should revisit what we gain out of re-using concludables instead of conclusions.
```

Comment:
Just trying to recall this right now, some historical possibilities that pop into mind:
- we didn't want to compute unification many times (not very expensive anyway)
- caching was done per-concludable and not per conclusion so reusing them was critical


Let's revisit this in 3.0